pandas/io/pickle.py

```diff
@@ -48,26 +69,43 @@ def try_read(path, encoding=None):
         # cpickle
         # GH 6899
```

Comment:
so these routines need to be changed to read in the file once e.g.

```
f, fh = _get_handle(....)
try:
buffer = BufferIO(fh.read())
finally:
   for _f in fh:
       f.close()  
```

then the operations are all

```
try:
    buffer.seek(0)
    pc.load(buffer....)
except:
    ...
````

etc, IOW, all we do is seek to the beginning of the buffer each time, rather than read the file in (potentially) 4 times.